js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+        };
+        if (since !== undefined) {
+            since = 0;
+            request['st'] = since;
+        }
+        if (limit !== undefined) {
+            request['count'] = limit;
+        }
+        const response = await this.v1GetMarketKline (this.extend (request, params));
+        const lists = this.safeValue (this.safeValue (response, 'data'), 'lists', []);
+        const ohlcvs = [];
+        for (let i = 0; i < lists.length; i++) {
+            const cur = lists[i];
+            const ohlcv = [parseInt (cur[6]) * 1000, cur[4], cur[2], cur[3], cur[1], cur[0]];  // columns - [ 'amount','close_price','max_price','min_price','open_price','pre_close_price','timestamp','usdt_amount','volume' ]
+            ohlcvs.push (ohlcv);
+        }
+        return this.parseOHLCVs (ohlcvs, market, timeframe, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        const timestamp = this.safeString (trade, 'timestamp');
+        const price = this.safeFloat (trade, 'deal_price');
+        const amount = this.safeFloat (trade, 'deal_volume');
+        const id = this.safeString (trade, 'id');
+        const side = undefined;
+        let cost = undefined;
+        const order = undefined;
+        let type = this.safeString (trade, 'order_type');
+        if (type !== undefined) {
+            type = this.options['order_type'][type];
+        }
+        if (price !== undefined) {
+            if (amount !== undefined) {
+                cost = amount * price;
+            }
+        }
+        const feeCost = this.safeFloat (trade, 'charge');
+        let fee = undefined;
+        const feeCurrency = undefined;
+        if (feeCost !== undefined) {
+            fee = {
+                'cost': feeCost,
+                'currency': feeCurrency,
+            };
+        }
+        let symbol = undefined;
+        if (market !== undefined) {
+            symbol = market['symbol'];
+        }
+        return {
+            'id': id,
+            'info': trade,
+            'order': order,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': type,
+            'side': side,
+            'takerOrMaker': undefined,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': fee,
+        };
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol_id': market['symbol_id'],
+        };
+        if (limit !== undefined) {
+            request['count'] = limit;
+        }
+        const response = await this.aicoinGetMarketAllTrade (this.extend (request, params));
+        const data = this.safeValue (this.safeValue (response, 'data', {}), 'list', []);
+        let result = [];
+        for (let i = 0; i < data.length; i++) {
```

Comment:
Instead of running this loop here, we need to return `this.parseTrades`, see how `fetchTrades` is implemented with the other exchanges – they all call `parseTrades` in the end and don't do the parsing inside the fetch-method itself. All parsing for public and private trades goes to `parseTrade`.